### PR TITLE
CMR-4998

### PR DESCRIPTION
**Pull request by ygliuvt**
Added support for UMM-S v1.2 in CMR.

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj [hunk 1]
```diff
@@ -41,6 +42,137 @@
...
+(defn- duplicate-supported-Formats
+  "Duplicate the SupportedFormats in v1.1 ServiceOptions to SupportedInputFormats
+  and SupportedOutputFormats in v1.2 ServiceOptions when migrate from 1.1 to 1.2."
+  [service-options]
+  (-> service-options
+      (assoc :SupportedInputFormats (:SupportedFormats service-options))
+      (assoc :SupportedOutputFormats (:SupportedFormats service-options))
+      (dissoc :SupportedFormats)))
+
+(def v1-1-supported-formats-enum->v1-2-supported-formats-enum
```
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj]: When reading the above changes, I was curious how you would address the change from mime-type to string; nice and simple -- I like it :-)

File: umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj [hunk 2]
```diff
@@ -66,3 +198,23 @@
       (update :Coverage migrate-coverage-type-down)
       (dissoc :RelatedURLs)
       util/remove-empty-maps))
+
```
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj]: Really nice job with all the migration code; I think this is the best example for migrations that I've seen in the CMR to-date.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj (hunk 1 above)
Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj]: Thanks, @oubiwann

File: umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj (hunk 2 above)
Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj]: Thanks!

File: umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj [hunk 3]
```diff
@@ -41,6 +42,137 @@
...
+  and SupportedOutputProjections in v1.2 ServiceOptions when migrate from 1.1 to 1.2."
+  [service-options]
+  (-> service-options
+      (assoc :SupportedInputProjections (seq (map #(hash-map :ProjectionName %)
+                                                  (:SupportedProjections service-options))))
+      (assoc :SupportedOutputProjections (seq (map #(hash-map :ProjectionName %)
+                                                   (:SupportedProjections service-options))))
+      (dissoc :SupportedProjections)))
+
+(defn- duplicate-supported-Formats
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj]: Nit: capital F

Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/service.clj]: Done. Thanks.

**Pull request merged**